mso/resource_mso_schema.go

```diff
@@ -80,25 +134,23 @@ func resourceMSOSchemaImport(d *schema.ResourceData, m interface{}) ([]*schema.R
 	if err != nil {
 		return nil, fmt.Errorf("No Template found")
 	}
-	found := false
+	templates := make([]interface{}, 0)
 	for i := 0; i < count; i++ {
-		tempCont, err := con.ArrayElement(i, "templates")
-
+		templatesCont, err := con.ArrayElement(i, "templates")
 		if err != nil {
-			return nil, fmt.Errorf("Unable to parse the template list")
+			return nil, fmt.Errorf("Unable to parse the templates list")
 		}
-		apiTemplate := models.StripQuotes(tempCont.S("name").String())
-		apiTenant := models.StripQuotes(tempCont.S("tenantId").String())
-		d.Set("template_name", apiTemplate)
-		d.Set("tenant_id", apiTenant)
-		found = true
-		break
+		map_template := make(map[string]interface{})
+		map_template["name"] = models.StripQuotes(templatesCont.S("name").String())
+		map_template["display_name"] = models.StripQuotes(templatesCont.S("displayName").String())
+		map_template["tenant_id"] = models.StripQuotes(templatesCont.S("tenantId").String())
+		templates = append(templates, map_template)
 
 	}
-	if !found {
-		d.Set("template_name", "")
-		d.Set("tenant_id", "")
-	}
+	d.Set("template", templates)
+	d.Set("template_name", "")
+	d.Set("tenant_id", "")
```

Comment:
Should we set those if it is a single template?

Reply:
Yes. When we apply a configuration using multiple templates, TF state always has "template_name", "" and "tenant_id", "" set as empty.
So it is good to have those on import as well. There won't be any issues when the imported configuration is reapplied.